plugin/forward/persistent.go

```diff
@@ -26,17 +28,21 @@ type transport struct {
 	yield chan *dns.Conn
 	ret   chan *dns.Conn
 	stop  chan bool
+
+	avgDialTime int64 // kind of average time of dial time
+
 }
 
 func newTransport(addr string, tlsConfig *tls.Config) *transport {
 	t := &transport{
-		conns:  make(map[string][]*persistConn),
-		expire: defaultExpire,
-		addr:   addr,
-		dial:   make(chan string),
-		yield:  make(chan *dns.Conn),
-		ret:    make(chan *dns.Conn),
-		stop:   make(chan bool),
+		conns:       make(map[string][]*persistConn),
+		expire:      defaultExpire,
+		addr:        addr,
+		dial:        make(chan string),
+		yield:       make(chan *dns.Conn),
+		ret:         make(chan *dns.Conn),
+		stop:        make(chan bool),
+		avgDialTime: 0,
```

Comment:
the default is 0?

Reply:
hum.. good point. Yes I want to be 0 to start with the right initialDialTime. But I should take care when computing the avg...